cmd/determinize-prow-config/main.go

```diff
@@ -181,3 +198,54 @@ func updatePluginConfig(configDir string) error {
 
 	return ioutil.WriteFile(configPath, data, 0644)
 }
+
+// prowConfigWithPointers mimics the upstream prowConfig but has pointer fields only
+// in order to avoid serializing empty structs.
+type prowConfigWithPointers struct {
```

Comment:
So it turns out we do need a different type, otherwise we end up marshalling a bunch of empty config in every file. This will get more complicated when we have this issue with substructs.

Reply:
Isn't this already an issue if we didn't have any BP for instance? Why not change upstream?

Reply:
Because changing upstream means to change a lot of code as now we have to nilcheck everywhere and it might end up sparkling a lot of debate.